src/stores/GuildMemberRoleStore.js

```diff
@@ -137,7 +148,11 @@ class GuildMemberRoleStore extends Collection {
    *   .then(member => console.log(`Member roles is now of ${member.roles.size} size`))
    *   .catch(console.error);
    */
-  set(roles, reason) {
+  set(roles = [], reason) {
+    if (!(roles instanceof Collection)) {
+      roles = new Collection(roles.map(role => [role, this.guild.roles.resolve(role)]));
```

Comment:
RoleResolvable is a Role|string, this would make it work if you pass an array of role objects
```suggestion
      roles = new Collection(roles.map(role => [this.guild.roles.resolveID(role), this.guild.roles.resolve(role)]));
```

Reply:
This doesn't matter as the original .set() method did absolutely nothing and just returned member.edit({roles})

Checking into member.edit itself shows that it doesn't use the keys of the collection, only the values, in fact actually if we wanted, we could change the ``this.guild.roles.resolve(role)`` into the .resolveID since member.edit() does this already ``if (data.roles) data.roles = data.roles.map(role => role instanceof Role ? role.id : role);`` but I wouldn't bother making another commit because it doesn't really matter either way